### ISSUE TITLE
Ignore all PMs

### ISSUE THREAD

**Issue by poixen** [Cockatrice/Cockatrice#1250]
As a user I want to be able to ignore all PMs to me.

**Comment by ZeldaZach** [comment]
Sounds reasonable, but there should be a mod/admin override on this so a staff member can always PM you

**Comment by Daenyth** [comment]
I don't think so; we don't allow mods to bypass direct ignore.

If you ignore a mod and you do something that gets a mod involved then you
deserve what you get imo

**Comment by poixen** [comment]
I agree. Mods are not gods. If you ignore a mod then you should be able to. If a mod REALLY needs to communicate, they can do so with bans.

**Comment by ZeldaZach** [comment]
I have this working client sided (You won't get any PMs sent to you, but the receiver will still think they went through).

I'm going to see if I can find a way to inform the sender that "This user isn't accepting PMs"

**Comment by Daenyth** [comment]
I don't think that's productive.

**Comment by woogerboy21** [comment]
@Daenyth whats not productive?  Ignoring all PMs or informing the other user?

**Comment by Daenyth** [comment]
Informing the other user.

**Comment by ZeldaZach** [comment]
Can I have more insight on why telling users "This user has private messaged disabled" is a bad thing @Daenyth ?

**Comment by Daenyth** [comment]
Because the person doing the ignoring probably doesn't want that fact broadcast. It leads to other users stalking them into games and main chat to bypass the pm block.

**Comment by Fenrirthviti** [comment]
Recent UI/UX paradigms have shown that sending an ambiguous message that the message could not be delivered are perfectly acceptable, and every major chat client (see: Discord as the prime example these days) has such a message. It's better to inform the user there was an error than silently ignore, IMO. A user who doesn't want to advertise they are blocking PMs is slightly suspicious in itself, and could lead to further issues (see: "Mods are not gods" comment, so if a mod is trying to reach a user who is blocking PMs, and the mod is never informed...)

For reference: 

![img](https://pub.rachni.com/img/2020-11-24_03-10-07.png)

Additionally, this brings up the idea for an option to only allow PMs from friends.
